airflow/providers/amazon/aws/sensors/sqs.py

```diff
@@ -215,3 +226,65 @@ def __init__(self, *args, **kwargs):
             stacklevel=2,
         )
         super().__init__(*args, **kwargs)
+
+
+class SqsBatchSensor(SqsSensor):
+    """
+    Get messages from an Amazon SQS queue in batches and then delete the retrieved messages from the queue.
+    If deletion of messages fails an AirflowException is thrown. Otherwise, all messages
+    are pushed through XCom with the key ``messages``.
+    The total number of messages retrieved at maxium will be equal to the number of messages retrived for each
+    SQS's API call multiplies with total number of call. Each SQS receive_message can get a max 10 messages.
+    This sensor is identical to SQSSensor, except the fact that SQSSensor performs one and only one SQS call
+    per poke, while SQSBatchSensor performs multiple SQS API calls per poke.
+    .. seealso::
+        For more information on how to use this sensor, take a look at the guide:
+        :ref:`howto/sensor:SqsBatchSensor`
+    :param batch: The number of time the sensor will call the SQS to receive messages (default: 1)
```

Comment:
Suggested rewording of the docstring.  It's still a bit confusing, but may be a bit better?

```suggestion
    Get messages from an Amazon SQS queue in batches and then delete the retrieved messages from the queue.
    If deletion of messages fails an AirflowException is thrown. Otherwise, all messages
    are pushed through XCom with the key ``messages``.
    
    This sensor is identical to SqsSensor, except that SqsSensor performs one and only one SQS call
    per poke, which limits the result to a maximum of 10 messages, while SqsBatchSensor performs multiple SQS API calls per poke and combines the results into one list.
    
    .. seealso::
        For more information on how to use this sensor, take a look at the guide:
        :ref:`howto/sensor:SqsBatchSensor`
        
    :param batch: The number of times the sensor will call the SQS API to receive messages (default: 1)
```

Reply:
Also, maybe add a line to the SqsSensor around L38, something like "The SQS API can only return up to 10 messages in a single call.  If a larger batch is needed, consider using SqsBatchSensor."